### PR TITLE
Support multi column results with using mongo query runner

### PR DESCRIPTION
**Pull request by del-zhenwu**
rebase from the latest master: https://github.com/getredash/redash/pull/4836/files

### CODE REVIEW

**Comment by justinclift** [comment]
@del-zhenwu Thanks for getting this rebased and resubmitted.

It's currently failing the lint/formatting check.  That should (in theory) be fairly easy to fix with the `make format` command, as long as things are setup according to the wiki:

https://github.com/getredash/redash/wiki/Local-development-setup#configuring-pre-commit

Hopefully that's useful? :smile:

**Comment by justinclift** [comment]
@del-zhenwu Cool, looks like it's passing most of the CI tests now, apart from the Code coverage check.

Are you ok to add some tests to this PR, to increase the coverage?

We don't *have* to, but if it's something that's easy to do then we probably should. :smile:

**Comment by del-zhenwu** [comment]
It seems no need to add ut, cause it just fixed the bug with no extra feature.

**Comment by justinclift** [comment]
@guidopetri @wlach This PR looks like it's ready to merge, but doesn't seem to have been reviewed yet (nor in it's previous PR).

Would either of you be ok to review and merge this as appropriate? :smile:

**Comment by justinclift** [comment]
@del-zhenwu Is there no way we can add tests for this?

It seems like we're missing an opportunity to improve our code if we don't. :wink:

File: redash/query_runner/mongodb.py [hunk 1]
```diff
@@ -93,39 +93,34 @@ def _get_column_by_name(columns, column_name):
...
     columns = []
 
     for row in results:
         parsed_row = {}
 
-        for key in row:
-            if isinstance(row[key], dict):
-                for inner_key in row[key]:
-                    column_name = "{}.{}".format(key, inner_key)
-                    if _get_column_by_name(columns, column_name) is None:
```
Comment (wlach) [inline redash/query_runner/mongodb.py:107]: I think you can remove `_get_column_by_name` now

**Review by wlach — CHANGES_REQUESTED** [review]
Hi @del-zhenwu, as @justinclift suggested, could you please add a small test case to the tests which demonstrates your fix? It should not take long, there is prior art for you to work off of:

https://github.com/getredash/redash/blob/650ec90df3441bdf8844ef741fb235ade6d3194e/tests/query_runner/test_mongodb.py#L122

File: redash/query_runner/mongodb.py (hunk 1 above)
Comment (del-zhenwu) [reply redash/query_runner/mongodb.py:107]: I will try.

Comment (del-zhenwu) [reply redash/query_runner/mongodb.py:107]: It has been used in unit test cases, I will keep it currently.

**Comment by del-zhenwu** [comment]
I updated the unit test case, to show it works in that case

File: redash/query_runner/mongodb.py (hunk 1 above)
Comment (guidopetri) [reply redash/query_runner/mongodb.py:107]: It seems it's still used in this file in the new line 339, old line 344.

**Review by guidopetri — APPROVED** [review]
This looks good to me! Thanks @del-zhenwu . I'm going to re-run the `make format` command so that it passes the lint test + merge `master` in.

**Comment by guidopetri** [comment]
I seem to have done something wrong with git, and now I can't edit your `master` branch anymore to reopen this PR @del-zhenwu . I'm going to open a new one with the same contents

**Comment by justinclift** [comment]
Oh.  It sounds like the behaviour of git being trigger happy to close PRs has some really bad consequences when the PR branch is in an external repo. :(

Or GitHub needs to fix their code, so PRs closed by their automated stuff allow for restoring a branch so they can still be worked on.

**Pull request closed without merging**